### ISSUE TITLE
license based testkube setup for public clouds

### ISSUE THREAD

**Issue by emamihe** [kubeshop/testkube#4219]
I recall a previous discussion about terraforming our infrastructure setup for users who wish to configure their own platforms like AWS, GCP, or Azure. I remember coming across a similar setup in the ROSA project, where Redhat's Openshift service on AWS could be deployed by providing an IAM token to Redhat on their website. They would handle the entire infrastructure setup in your environment. The monetization model was based on licensing per computing power and also included SRE support for Openshift by Redhat. All the associated costs, such as licenses and SRE support, would be listed on, let's say, the AWS bill. I believe this model could also be applicable to Testkube for enterprise users. Thus, I'd like to share this idea for a brainstorming session to determine its feasibility for our organization.

CC:  @TheBrunoLopes @olensmar  @dejanzele

**Comment by rangoo94** [comment]
When it comes to Enterprise, we already have an automated setup, that simplifies the procedure of starting the Enterprise application. That issue is only related then to the Open Source version (including Cloud/Enterprise agents) then.

Setting up such Terraform for Testkube won't be that simple. We need to consider two cases: 1) users that want to use Testkube for simple tests with no effort, and 2) users that want to use Testkube to test in their infrastructure.

* For (1) users that want to use Testkube for simple tests, we are considering lambdas ([**KNative**](https://github.com/kubeshop/testkube/issues/4178)), so running would be much easier.
* For (2) users that want to use Testkube in their infrastructure, it's impossible to set it up fully automatically
  * **Networking:** You need to set up IAM / VPC(E) / NATs / Gateways, that will be very specific for your organization
  * **Kubernetes:** If the User wants a separate cluster, it could be simple. The problem happens when the Kubernetes cluster is shared (to i.e. access internal applications by Testkube without exposing them out of the cluster)
  * **Testkube configuration:** even for a simple run, it would be very much dependent on the points from above

Basically, the only thing we could do is set up a Terraform module wrapping around the Helm chart. Such a module would be really small, but will not be used a lot, yet it would need to be maintained.

I feel that the effort would be big and constant, while it would not increase the Testkube adoption.
